invoice_comment_template/models/account_invoice.py

```diff
@@ -27,3 +27,13 @@ def _set_note2(self):
         comment = self.comment_template2_id
         if comment:
             self.note2 = comment.get_value(self.partner_id.id)
+
+    @api.onchange('partner_id', 'company_id')
+    def _onchange_partner_id(self):
+        res = super(AccountInvoice, self)._onchange_partner_id()
+        comment_template = self.partner_id.comment_template_id
+        if comment_template.position == 'before_lines':
```

Comment:
Don't perform anything if not `comment_template`

Reply:
It's removed template_id if there is not not comment_template